### PR TITLE
Wearing a pirate hat now changes your speech to be piratey

### PR DESCRIPTION
**Pull request by ghost**
This be a great idea, matey

##### Changelog

:cl:
rscadd: Wearing a pirate hat will now change your speech to be piratey.  Yarr.
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/misc.dm [hunk 1]
```diff
@@ -118,19 +118,130 @@
...
 	name = "pirate hat"
 	desc = "Yarr."
 	icon_state = "hgpiratecap"
 	item_state = "hgpiratecap"
+	
+/obj/item/clothing/head/hgpiratecap/speechModification(message)
+	if(copytext(message, 1, 2) != "*")
+		message = replacetext(message, "captain", "CAP'N")
+		message = replacetext(message, "station", "SHIP")
+		message = replacetext(message, "sir", "sir")
```
Comment (X-TheDark) [inline code/modules/clothing/head/misc.dm:169]: Nice replacetext No.1

File: code/modules/clothing/head/misc.dm [hunk 2]
```diff
@@ -118,19 +118,130 @@
...
+		message = replacetext(message, "there", "tharr")
+		message = replacetext(message, "my", "me")
+		message = replacetext(message, "is", "be")
+		message = replacetext(message, "it's", "be")
+		message = replacetext(message, "friend", "matey")
+		message = replacetext(message, "buddy", "matey")
+		message = replacetext(message, "hello", "ahoy")
+		message = replacetext(message, "hot", "swelterin'")
+		message = replacetext(message, "very", "extra")
+		message = replacetext(message, "sword", "sword")
```
Comment (X-TheDark) [inline code/modules/clothing/head/misc.dm:179]: Nice replacetext No.2

File: code/modules/clothing/head/misc.dm [hunk 3]
```diff
@@ -118,19 +118,130 @@
...
 	name = "pirate bandana"
 	desc = "Yarr."
 	icon_state = "bandana"
 	item_state = "bandana"
 
+/obj/item/clothing/head/bandana/speechModification(message)
+	if(copytext(message, 1, 2) != "*")
+		message = replacetext(message, "captain", "CAP'N")
+		message = replacetext(message, "station", "SHIP")
+		message = replacetext(message, "sir", "sir")
```
Comment (X-TheDark) [inline code/modules/clothing/head/misc.dm:212]: Nice replacetext No.3

File: code/modules/clothing/head/misc.dm [hunk 4]
```diff
@@ -118,19 +118,130 @@
...
+		message = replacetext(message, "there", "tharr")
+		message = replacetext(message, "my", "me")
+		message = replacetext(message, "is", "be")
+		message = replacetext(message, "it's", "be")
+		message = replacetext(message, "friend", "matey")
+		message = replacetext(message, "buddy", "matey")
+		message = replacetext(message, "hello", "ahoy")
+		message = replacetext(message, "hot", "swelterin'")
+		message = replacetext(message, "very ", "extra")
+		message = replacetext(message, "sword", "sword")
```
Comment (X-TheDark) [inline code/modules/clothing/head/misc.dm:222]: Nice replacetext No.4

**Review by X-TheDark — CHANGES_REQUESTED** [review]
Nice replacetexts, Adam. Very quality. Much good.
Other than that, I approve.

File: code/modules/clothing/head/misc.dm (hunk 4 above)
Comment (ghost) [reply code/modules/clothing/head/misc.dm:222]: This is literally how it is done in the luchador mask code.

**Comment by X-TheDark** [comment]
George is probably going to be furious.

**Comment by ghost** [comment]
I'm still confused.

Replacetext is how it's done in the luchador mask code.

Aside from me hitting S and not realizing it a couple of times what's the problem.

**Comment by X-TheDark** [comment]
`speechModification` proc is only defined for **masks**, aka, those things that have `/obj/item/clothing/mask/` in their typepath.

**Comment by ghost** [comment]
Why isn't it defined for clothes.

Great.

Well, do I have to rewrite the proc entirely for this?

**Comment by Super3222** [comment]
Stop using the online editor.
and to answer your question... yes. You would probably need to add a new variable to clothing for changing speech, and have the say() proc check for it. otherwise, the safe route is making pirate bandana (mask version) only.

**Comment by ghost** [comment]
@Super3222 Open this you numbnuts i'm not done

**Comment by ghost** [comment]
@Super3222 YOU SHIT OPEN THIS

**Comment by Super3222** [comment]
You have other open PR's that you need to work on.
As for this one, open a new PR with these changes, and make sure it's under a git program. Not the online editor.

**Pull request closed without merging**